stdlib/2.7/_weakrefset.pyi

```diff
@@ -1,5 +1,14 @@
-from typing import Iterator, Any
+from typing import Iterator, Any, Iterable, MutableSet, TypeVar, Generic
 
-class WeakSet:
-    def __iter__(self) -> Iterator[Any]: ...
-    def add(self, *args: Any, **kwargs: Any) -> Any: ...
+_T = TypeVar('_T')
+
+class WeakSet(MutableSet[_T], Generic[_T]):
+    def __init__(self, data: Iterable[_T] = ...) -> None: ...
+
+    def add(self, x: _T) -> None: ...
+    def discard(self, x: _T) -> None: ...
+    def __contains__(self, x: Any) -> bool: ...
```

Comment:
You shouldn't need `__contains__` and following -- they are inherited from MutableSet. You only need add and discard here because those are abstract there.